pkg/operator/ceph/cluster/osd/pod.go

```diff
@@ -39,65 +42,148 @@ const (
 	osdWalSizeEnvVarName        = "ROOK_OSD_WAL_SIZE"
 	osdJournalSizeEnvVarName    = "ROOK_OSD_JOURNAL_SIZE"
 	osdMetadataDeviceEnvVarName = "ROOK_METADATA_DEVICE"
+	osdPrepareOnlyEnvVarName    = "ROOK_OSD_PREPARE_ONLY"
 )
 
-func (c *Cluster) makeDaemonSet(selection rookalpha.Selection, storeConfig config.StoreConfig, metadataDevice, location string) *extensions.DaemonSet {
-	podSpec := c.podTemplateSpec(nil, selection, c.resources, storeConfig, metadataDevice, location)
-	return &extensions.DaemonSet{
+func (c *Cluster) makeJob(nodeName string, devices []rookalpha.Device,
+	selection rookalpha.Selection, resources v1.ResourceRequirements, storeConfig config.StoreConfig, metadataDevice, location string) *batch.Job {
+
+	podSpec := c.podTemplateSpec(devices, selection, resources, storeConfig, metadataDevice, location, true /* prepare-only */, v1.RestartPolicyOnFailure)
+	podSpec.Spec.NodeSelector = map[string]string{apis.LabelHostname: nodeName}
+
+	return &batch.Job{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:            appName,
+			Name:            fmt.Sprintf(prepareAppNameFmt, nodeName),
 			Namespace:       c.Namespace,
 			OwnerReferences: []metav1.OwnerReference{c.ownerRef},
 			Labels: map[string]string{
-				k8sutil.AppAttr:     appName,
+				k8sutil.AppAttr:     prepareAppName,
 				k8sutil.ClusterAttr: c.Namespace,
 			},
 		},
-		Spec: extensions.DaemonSetSpec{
-			UpdateStrategy: extensions.DaemonSetUpdateStrategy{
-				Type: extensions.RollingUpdateDaemonSetStrategyType,
-			},
+		Spec: batch.JobSpec{
 			Template: podSpec,
 		},
 	}
 }
 
-func (c *Cluster) makeReplicaSet(nodeName string, devices []rookalpha.Device, selection rookalpha.Selection, resources v1.ResourceRequirements,
-	storeConfig config.StoreConfig, metadataDevice, location string) *extensions.ReplicaSet {
-
-	podSpec := c.podTemplateSpec(devices, selection, resources, storeConfig, metadataDevice, location)
-	podSpec.Spec.NodeSelector = map[string]string{apis.LabelHostname: nodeName}
+func (c *Cluster) makeOSDDeployment(nodeName string, devices []rookalpha.Device, selection rookalpha.Selection, resources v1.ResourceRequirements,
```

Comment:
I think the `OSD` in the function name is redundant here because of the package name, I would make it `makeDeployment`.
When looking at the previous function name too